test/404s.test.js

```diff
@@ -211,6 +211,39 @@ test('setting a custom 404 handler multiple times is an error', t => {
       fastify.close()
     })
   })
+
+  t.test('in separate plugins at the same level', t => {
+    t.plan(3)
+
+    const fastify = Fastify()
+
+    fastify.register((instance, options, next) => {
+      instance.register((instance2A, options, next) => {
+        instance2A.setNotFoundHandler(() => {})
+        next()
+      })
+
+      instance.register((instance2B, options, next) => {
+        try {
+          instance2B.setNotFoundHandler(() => {})
```

Comment:
I don't agree here. Two different plugins can have different 404 handler.

A plugin doesn't know where is mounted. This breaks plugin encapsulation.

Reply:
If the two plugins each had their own 404 handler, what two URLs could you use match each of the handlers?